syntax = "proto3";
package VRDAVis;

import "enums.proto";

// REGISTER_VIEWER:
// Registers the viewer with the backend.
// Responds with REGISTER_VIEWER_ACK
message RegisterViewer {
    fixed32 session_id = 1;
    // string api_key = 2;
}

// REGISTER_VIEWER_ACK
// Acknowledgement response for REGISTER_VIEWER.
// Informs the frontend whether the session was correctly.
message RegisterViewerAck {
    // Unique session ID
    fixed32 session_id = 1;
    // Defines whether the REGISTER_VIEWER command was successful
    bool success = 2;
    // Error message (if applicable)
    string message = 3;
}